cli/src/cmd_auth.rs

```diff
@@ -535,6 +519,78 @@ impl CmdAuthStatus {
     }
 }
 
+/// Rename a profile.
+///
+/// This command does not modify the profile credentials. If
+/// the profile being renamed is your `default-profile` that
+/// will be updated as well.
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+pub struct CmdAuthRenameProfile {
+    /// The current name of the profile to be renamed.
+    current_name: String,
+
+    /// The new name of the profile to be renamed.
+    new_name: String,
```

Comment:
nothing else in the CLI has positional arguments; I'm loath to start here. What do you think? I think I'll often end up looking up whether this uses `memcpy` or `bcopy` semantics...

e.g. --from / --to or --profile / --new-name